src/inmanta/deploy/scheduler.py

```diff
@@ -97,50 +105,73 @@ async def new_version(
                 if dropped:
                     self._state.update_pending.add(resource)
                     dropped_requires[resource] = dropped
-
+                # this loop is race-free, potentially slow, and completely synchronous
+                # => regularly pass control to the event loop to not block scheduler operation during update prep
+                await asyncio.sleep(0)
```

Comment:
this is a point to keep in mind, once for every resource might be too much, otoh, it might be perfect